### PR TITLE
Update TensorFlow to 2.x, and add model checkpoints, add image extension option

### PR DESCRIPTION
**Pull request by mgagvani**
1. Currently TensorKart uses TF 1.x, while most code has transitioned to 2.x. The only changes needed were to the imports (since Keras is part of TensorFlow in 2.x). 

2. Added a model checkpoint, which will only save models with an improved (lower) validation loss during training. The previous approach of just saving the last model usually resulting in a very overfit model being saved for me. 

3. Added variable `IMAGE_TYPE` to `record.py`. By default, it is set to `.png`. However, increasing the resolution of the saved images would cause the true sampling rate to become far lower than the rate specified in the code. Changing the extension to `.jpg` fixed this issue.

(By the way, these changes are coming from my ["experimentation" fork](https://github.com/mgagvani/Xbox-Game-AI) of TensorKart which contains a lot of extra code for end-to-end Windows game control).

### CODE REVIEW

**Comment by kevinhughes27** [comment]
> increasing the resolution of the saved images would cause the true sampling rate to become far lower than the rate specified in the code

Is this because the png file is larger and writing to disk takes longer?

**Comment by mgagvani** [comment]
Yes.

**Review by kevinhughes27 — APPROVED** [review]
Thanks for the PR!

I was also wondering if you'd like to start a `Related Projects` section in the readme and we can start adding some links in there. I was just checking out your https://github.com/mgagvani/Xbox-Game-AI project and it looks pretty cool!

**Comment by mgagvani** [comment]
I think the `Related Projects` section is a cool idea, where should it go in the README? After `Future Work / Ideas`?

**Comment by kevinhughes27** [comment]
Yeah I agree after `Future Work / Ideas` seems like the right spot :)
